src/GafferUI/PlugAdder.cpp

```diff
@@ -0,0 +1,354 @@
+
+IE_CORE_DEFINERUNTIMETYPED( PlugAdder );
+
+PlugAdder::PlugAdder( Gaffer::NodePtr node, StandardNodeGadget::Edge edge )
+	:	m_node( node ), m_edge( edge ), m_dragging( false )
+{
+	node->childAddedSignal().connect( boost::bind( &PlugAdder::childAdded, this ) );
+	node->childRemovedSignal().connect( boost::bind( &PlugAdder::childRemoved, this ) );
+
+	enterSignal().connect( boost::bind( &PlugAdder::enter, this, ::_1, ::_2 ) );
+	leaveSignal().connect( boost::bind( &PlugAdder::leave, this, ::_1, ::_2 ) );
+	buttonPressSignal().connect( boost::bind( &PlugAdder::buttonPress, this, ::_1,  ::_2 ) );
+	dragBeginSignal().connect( boost::bind( &PlugAdder::dragBegin, this, ::_1, ::_2 ) );
+	dragEnterSignal().connect( boost::bind( &PlugAdder::dragEnter, this, ::_2 ) );
+	dragMoveSignal().connect( boost::bind( &PlugAdder::dragMove, this, ::_1, ::_2 ) );
+	dragLeaveSignal().connect( boost::bind( &PlugAdder::dragLeave, this, ::_2 ) );
+	dropSignal().connect( boost::bind( &PlugAdder::drop, this, ::_2 ) );
+	dragEndSignal().connect( boost::bind( &PlugAdder::dragEnd, this, ::_2 ) );
+
+	updateVisibility();
+}
+
+PlugAdder::~PlugAdder()
+{
+}
+
+Imath::Box3f PlugAdder::bound() const
+{
+	return Box3f( V3f( -0.5f, -0.5f, 0.0f ), V3f( 0.5f, 0.5f, 0.0f ) );
+}
+
+void PlugAdder::updateDragEndPoint( const Imath::V3f position, const Imath::V3f &tangent )
+{
+	m_dragPosition = position;
+	m_dragTangent = tangent;
+	m_dragging = true;
+	requestRender();
+}
+
+void PlugAdder::doRender( const Style *style ) const
+{
+	if( m_dragging )
+	{
+		if( !IECoreGL::Selector::currentSelector() )
+		{
+			V3f srcTangent( 0.0f, 0.0f, 0.0f );
+			style->renderConnection( V3f( 0 ), srcTangent, m_dragPosition, m_dragTangent, Style::HighlightedState );
+		}
+	}
+
+	float radius = 0.75f;
+	Style::State state = Style::NormalState;
+	if( getHighlighted() )
+	{
+		radius = 1.25f;
+		state = Style::HighlightedState;
+	}
+	style->renderImage( Box2f( V2f( -radius ), V2f( radius ) ), texture( state ) );
+}
+
+void PlugAdder::addPlug( Gaffer::Plug *connectionEndPoint )
+{
+	UndoContext undoContext( m_node->ancestor<ScriptNode>() );
+
+	if( SwitchComputeNode *switchNode = runTimeCast<SwitchComputeNode>( m_node.get() ) )
+	{
+		switchNode->setup( connectionEndPoint );
+		ArrayPlug *inPlug = switchNode->getChild<ArrayPlug>( "in" );
+		Plug *outPlug = switchNode->getChild<Plug>( "out" );
+
+		StandardNodeGadget::Edge inEdge = StandardNodeGadget::InvalidEdge;
+		if( connectionEndPoint->direction() == Plug::Out )
+		{
+			inPlug->getChild<Plug>( 0 )->setInput( connectionEndPoint );
+			inEdge = m_edge;
+		}
+		else
+		{
+			connectionEndPoint->setInput( switchNode->getChild<Plug>( "out" ) );
+			inEdge = oppositeEdge( m_edge );
+		}
+
+		updateMetadata( inPlug, "nodeGadget:nodulePosition", edgeName( inEdge ) );
+		updateMetadata( inPlug, "compoundNodule:orientation", edgeOrientation( inEdge ) );
+		updateMetadata( outPlug, "nodeGadget:nodulePosition", edgeName( oppositeEdge( inEdge ) ) );
```

Comment:
Wait, maybe I've got that backwards and the "funky" colors are the ones coming from the IE config, which were rarely seen before, since color and float nodules are exposed almost exclusively by shaders... so maybe we should just update our coding to use the same colors that gaffer is setting for shader plugs?

It still would be nice to have the instance colors from the Jabuka nodes transfer over though...